privacy/lmt/ios.go

```diff
@@ -0,0 +1,67 @@
+package lmt
+
+import (
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/iosutil"
+)
+
+var (
+	int8Zero int8 = 1
+	int8One  int8 = 1
+)
+
+// ModifyForIOS modifies the request's LMT flag based on iOS version and identity.
+func ModifyForIOS(req *openrtb.BidRequest) {
+	modifiers := map[iosutil.VersionClassification]modifier{
+		iosutil.Version140:          modifyForIOS14X,
+		iosutil.Version141:          modifyForIOS14X,
+		iosutil.Version142OrGreater: modifyForIOS142OrGreater,
+	}
+	modifyForIOS(req, modifiers)
+}
+
+func modifyForIOS(req *openrtb.BidRequest, modifiers map[iosutil.VersionClassification]modifier) {
+	if !isRequestForIOS(req) {
+		return
+	}
+
+	versionClassification := iosutil.DetectVersionClassification(req.Device.OSV)
+	if modifier, ok := modifiers[versionClassification]; ok {
+		modifier(req)
+	}
```

Comment:
My first thought here was null object pattern but perhaps what you have is better.

Reply:
I love that the null object pattern was in your mind. :)

I don't think it applies in this situation since the modifier func isn't a return value passed between methods. I consider the range coverage of > 14.0 to be a coincidence which may change in the future. I did not intend for the map to be a comprehensive of all possible iOS versions.